e2e/workload_id/input/api-nomad-cli.nomad.hcl

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+job "task-api-nomad-cli" {
+  type = "batch"
+
+  group "grp" {
+    restart { attempts = 0 }
+    reschedule { attempts = 0 }
+    constraint {
+      attribute = "${attr.kernel.name}"
+      value     = "linux"
+    }
+
+    task "tsk" {
+      driver = "raw_exec"
+      config {
+        command = "bash"
+        // "|| true" because failure to get a var makes nomad cli exit 1,
+        // but for this test, "Variable not found" actually indicates successful
+        // API connection.
+        args = ["-xc", "echo $NOMAD_ADDR; nomad var get nothing || true"]
```

Comment:
Wouldn't any other error throw an error too? If we registered a Nomad-native service for this job, the default Workload Identity will give us access to read services and then we could verify it works end-to-end.

Reply:
I check Stderr in the test to make sure we specifically get "Variable not found" and not some other misc error. I'm not trying to test real workload identity behavior, but I suppose I could extend the scope a bit like this if you'd like.

Reply:
Oh yeah I see. Makes sense... and having the task fail would make it hard to differentiate between "something completely different failed".